cmd/helm/helm.go

```diff
@@ -249,6 +251,33 @@ func getKubeClient(context string) (*restclient.Config, *internalclientset.Clien
 	return config, client, nil
 }
 
+func loadAuthHeaders(ctx context.Context) context.Context {
+	c, err := kube.GetConfig(kubeContext).ClientConfig()
+	if err != nil {
+		log.Println("Failed to extract authentication headers")
+		return ctx
+	}
+
+	if c.AuthProvider != nil {
+		switch c.AuthProvider.Name {
+		case "gcp":
+			ctx = context.WithValue(ctx, kube.Authorization, "Bearer "+c.AuthProvider.Config["access_token"])
+		case "oidc":
+			ctx = context.WithValue(ctx, kube.Authorization, "Bearer "+c.AuthProvider.Config["id-token"])
+		default:
```

Comment:
panicing will cause a stack trace to be displayed to the end user. Do we really want that?